median/aggregated_attribute_observation.go

```diff
@@ -33,6 +34,12 @@ func aggregate(observations []median.ParsedAttributedObservation) *aggregatedAtt
 	})
 	aggregated.JuelsPerFeeCoin = observations[n/2].JuelsPerFeeCoin
 
+	// compatible with nil case
```

Comment:
what i mean here is that if in the nil case, the `aggregated.GasPriceSubunit` will be set to nil because you can compare two nils (returns 0) and the median entry will be nil since all the gas data points are nil

Reply:
I thought about adding a special error case for if you have a nil value and a non-nil value -- but decided not to. It could already happen with juelsperfeecoin or the ocr value and we just trust it to panic in that case.

Reply:
EDIT: correction, it is currently impossible to have nil observation values. But with https://github.com/smartcontractkit/offchain-reporting/pull/412/files#diff-f47f225a9c371e457320c9da5a88b0927ce8060ce4c02aa8ba40bf92f010cf74R347 only GasPriceSubunit would be possible to be nil.

Thinking through the edge cases,
Is it better to 
a. fail if gas prices are a mixture of nil and non-nil? (this is what currently happens)
b. transmit the median value

Worst case for scenario (a) is that transmit() is blocked until all nops agree on nil vs non-nil. This would affect all chains, including EVM chains if some nops are incorrectly set.

Worst case scenario for (b) is that only Starknet would be blocked if the median value is `nil` because for EVM chains the value of GasPrice would never be used regardless of whether a nil or non-nil is the final value.

I think it's safer to go with (b)? If so, I can adjust the code

Reply:
EDIT: fixed so that nil values are never allowed for gas price (like for julesperfeecoin and datasourcevalue). so all values will technically panic on nil, but the data source never submits a nil.